src/mono/wasm/debugger/BrowserDebugProxy/ValueOrError.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+
+namespace Microsoft.WebAssembly.Diagnostics;
+
+public struct ValueOrError<TValue>
+{
+    public TValue? Value { get; init; }
+    public Result? Error { get; init; }
+
+    public bool HasValue => Value != null;
```

Comment:
If the name includes *or* then can we please change to IsError and IsValue so that it will be clear that it cannot have both? Without the need to check what is in the constructor ;)